File: Difficulty: Medium/Last Moment Before All Ants Fall Out/last-moment-before-all-ants-fall-out.py
class Solution:
    # 24.07.25 potd
    def getLastMoment(self, n, left, right):
        # If 'left' is not empty, max time is the furthest left ant's position
        max_left = max(left) if left else 0
        
        # If 'right' is not empty, max time is the max(n - pos)
        max_right = max((n - r for r in right), default=0)
        
        # The result is the maximum time any ant takes to fall
        return max(max_left, max_right)

File: Difficulty: Medium/Last Moment Before All Ants Fall Out/last-moment-before-all-ants-fall-out.cpp
class Solution {
  public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        // code here
        // 04.02.26 POTD
        int ans = 0;
        for(int a: left){
            ans = max(ans, a);
        }
        for(int a: right){
            ans = max(ans, n-a);
        }
        return ans;
    }
};